recipe/meta.yaml

```
@@ -11,40 +11,58 @@ source:
 
 build:
   number: 0
-  noarch: python
+  skip: True  # [py<37]
   entry_points:
     - pip-compile = piptools.scripts.compile:cli
     - pip-sync = piptools.scripts.sync:cli
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install --no-deps --no-build-isolation . -vv
 
 requirements:
   host:
-    - python >=3.7
+    - python
     - setuptools-scm
+    - setuptools
     - pip
+    - wheel
   run:
-    - python >=3.7
+    - python
     - python-build
-    - click >=7
-    - pip >=21.2
+    - click >=8
+    - pip >=22.2
     - setuptools
     - wheel
+    - tomli  # [py<311]
 
 test:
+  source_files:
+    - tests
   imports:
     - piptools
   commands:
     - pip check
     - pip-compile --help
     - pip-sync --help
+    - pytest -vv tests -k "not (test_combine_different_extras_of_the_same_package or test_pass_pip_cache_to_pip_args)"
```

```suggestion
    - pytest -vv tests -k "not (test_combine_different_extras_of_the_same_package or test_pass_pip_cache_to_pip_args)" -m 'not network'
```
See https://github.com/jazzband/pip-tools/blob/main/pyproject.toml#L115